src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -11,15 +11,26 @@ public class ReadWriteList<E> {
 
     public void add(E element) {
         // write your code here
+        lock.writeLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public E get(int index) {
         // write your code here
-        return null;
+        lock.readLock().lock();
+        try {
+            return list.get(index);
+        } finally {
+            lock.readLock().unlock();
+        }
     }
 
     public int size() {
         // write your code here
-        return 0;
+        return list.size();
```

Comment:
The `size` method should be synchronized with a read lock to ensure thread safety. Consider using `lock.readLock().lock()` and `lock.readLock().unlock()` around the `return list.size();` statement.